### PR TITLE
Fix Ledger messages signatures

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/20864

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Go to snapshot.org
- Connect Ledger account and vote

### CODE REVIEW

File: components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.test.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ test('Sign personal message successfully', () => {
   ledgerHardwareKeyring.app.signature = { v: 1, r: 'b68983', s: 'r68983' }
   return expect(ledgerHardwareKeyring.signPersonalMessage(
     'm/44\'/60\'/0\'/0/0', 'message'))
-    .resolves.toStrictEqual({ payload: '0xb68983r68983-26', success: true })
+    .resolves.toStrictEqual({ payload: '0xb68983r68983-1a', success: true })
```
Comment (onyb) [inline components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.test.ts:117]: When `v` is `1`, the serialised hex should be `01` and not `-1a`. Let's also have another test case where `v` is `27` or `28`.

**Review by onyb — CHANGES_REQUESTED** [review]
Left some remarks about a possible bug. I found that Metamask's Ledger keyring has the same issue (see [source](https://github.com/MetaMask/eth-ledger-bridge-keyring/blob/438bc2d470c2302b94b01706671a08becc6b4a07/index.js#L340-L344)), but let's be defensive on our side.

File: components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.ts [hunk 2]
```diff
@@ -134,7 +134,7 @@ export default class LedgerBridgeKeyring extends LedgerEthereumKeyring {
   }
 
   private readonly createMessageSignature = (result: SignatureVRS) => {
-    let v = (result.v - 27).toString()
+    let v = (result.v - 27).toString(16)
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.ts]: Thanks for clarification @onyb, updated

File: components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.test.ts (hunk 1 above)
Comment (spylogsster) [reply components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.test.ts]: done

File: components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.ts [hunk 3]
```diff
@@ -134,7 +134,12 @@ export default class LedgerBridgeKeyring extends LedgerEthereumKeyring {
   }
 
   private readonly createMessageSignature = (result: SignatureVRS) => {
-    let v = (result.v - 27).toString()
+    // Convert the recovery identifier to standard ECDSA if using bitcoin secp256k1 convention.
+    let v = result.v < 27
+      ? result.v.toString(16)
+      : (result.v - 27).toString(16)
+
+    // Pad v with a leading zero if value is under `16` (i.e., single character hex).
```
Comment (darkdh) [inline components/brave_wallet_ui/common/hardware/ledgerjs/eth_ledger_bridge_keyring.ts:142]: please also add a test case that doesn't need padding

**Review by darkdh — APPROVED** [review]

**Pull request merged**